ci/bin/build-and-test

```diff
@@ -1,12 +1,28 @@
 #!/usr/bin/env ruby
 
 require "fileutils"
+require "pathname"
+require "yaml"
+
+def ensure_everything_is_committed
+  out = `git status -s`
+
+  raise StandardError, "there are uncommitted changes or un-staged files: #{out}" unless out.empty?
+end
+
+def build_rails_version_specifier(target_versions_path)
+  major_minor = YAML.safe_load(File.read(target_versions_path)).fetch("target_rails_major_minor")
+
+  "~> #{major_minor}.0"
+end
```

Comment:
I'm deliberately duplicating the logic for building the version specifier here and in `template.rb`. It's small and I don't want to make `template.rb` more complicated than it absolutely needs to be.